hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -82,6 +83,39 @@ object IRFunctionRegistry {
       })
   }
 
+  def pyRegisterIRForServiceBackend(
+    ctx: ExecuteContext,
+    name: String,
+    typeParamStrs: Array[String],
+    argNames: Array[String],
+    argTypeStrs: Array[String],
+    returnType: String,
+    bodyStr: String
+  ): Unit = {
```

Comment:
This is just a minor modification to the above function:
1. Directly consume Scala Arrays instead of ArrayList. 
2. Construct a refMap from the arguments and parameters and use that to parse the body.

Unlike the Py4J backends, in the service backend, the IR will never be already constructed in Java, so it makes sense to do all the type, argument, and body parsing in one method all together.